### PR TITLE
Add initial version of dreg

### PR DESCRIPTION
**Pull request by camh-**
Add a basic implementation of dreg to demonstrate `foxygo.at/protog/httprule`.
This implementation has a couple of basic commands and will grow as
functionality is added to `httprule`, such as handling request and response
headers, which are required for the docker registry API.

Hermit is used for the required tools that are in hermit, and `make tools` will
install the rest.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+# --- Lint ---------------------------------------------------------------------
+
+lint:  ## Lint go source code
+	golangci-lint run
+
+.PHONY: lint
+
+# --- Protos ---------------------------------------------------------------------
+
+protos:
```
Comment (juliaogris) [inline Makefile]: Add comment maybe.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+# --- Utilities ----------------------------------------------------------------
+COLOUR_NORMAL = $(shell tput sgr0 2>/dev/null)
+COLOUR_RED    = $(shell tput setaf 1 2>/dev/null)
+COLOUR_GREEN  = $(shell tput setaf 2 2>/dev/null)
+COLOUR_WHITE  = $(shell tput setaf 7 2>/dev/null)
+
+help:
+	@awk -F ':.*## ' 'NF == 2 && $$1 ~ /^[A-Za-z0-9%_-]+$$/ { printf "$(COLOUR_WHITE)%-25s$(COLOUR_NORMAL)%s\n", $$1, $$2}' $(MAKEFILE_LIST) | sort
+
+tools:
```
Comment (juliaogris) [inline Makefile]: maybe here too.

File: proto/registry.proto [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  string name = 1;
+  repeated string tags = 2;
+}
+
+message DeleteImageRequest {
+  string name = 1;
+  string reference = 2;
+}
+
+message DeleteImageResponse {}
```
Comment (juliaogris) [inline proto/registry.proto:57]: This file is really, really neat. so cool how obvious (the whole file).

**Review by juliaogris — APPROVED** [review]
I miss reviewing your code - reads sooo well.
🟢

File: Makefile (hunk 1 above)
Comment (camh-) [reply Makefile]: Done

File: Makefile (hunk 2 above)
Comment (camh-) [reply Makefile]: Done

File: proto/registry.proto (hunk 3 above)
Comment (camh-) [reply proto/registry.proto:57]: :sandwich:

**Pull request merged**